player/lua/ytdl_hook.lua

```diff
@@ -626,9 +626,13 @@ mp.add_hook(o.try_ytdl_first and "on_load" or "on_load_fail", 10, function ()
                     site = entry["webpage_url"]
                 end
 
-                -- links with only youtube id as returned by --flat-playlist
+                -- links without protocol as returned by --flat-playlist
                 if not site:find("://") then
-                    table.insert(playlist, "https://youtu.be/" .. site)
+                    -- youtube extractor provides only IDs,
+                    -- others come prefixed with the extractor name and ":"
+                    local prefix = site:find(":") and "ytdl://" or
+                        "https://youtu.be/"
```

Comment:
how sure are we that youtube is the only place we see such urls?

Reply:
Not sure. Would need to go through youtube-dl's url parsing, but just looking at youtube's extractor, everything except the ID is optional. Looking at lazy_extractors.py, which is used to generate an optional list of valid URLs for faster extraction, there's 35 extractors which use the shortcut "<extractor>:<id>". There's no such shortcut for youtube.